packages/@tinacms/react-sidebar/src/components/Sidebar.tsx

```diff
@@ -27,47 +27,67 @@ import {
   TinaIcon,
 } from '@tinacms/icons'
 import { TinaReset } from '@tinacms/styles'
-import { SIDEBAR_WIDTH, SIDEBAR_HEADER_HEIGHT } from '../../Globals'
+import { SIDEBAR_WIDTH, SIDEBAR_HEADER_HEIGHT } from './Globals'
 import { CreateContentMenu } from './CreateContentMenu'
 import { ScreenPlugin, ScreenPluginModal } from '@tinacms/react-screens'
-import { useSubscribable, useCMS } from '../../react-tinacms'
-import { SidebarState, SidebarPosition } from '../../tina-cms'
+import { useSubscribable, useCMS } from '@tinacms/react-core'
+import { SidebarState, SidebarPosition, SidebarStateOptions } from '../sidebar'
 
-export function SidebarProvider({ children, position, hidden, cms }: any) {
-  useSubscribable(cms.sidebar)
+export interface SidebarProviderProps {
+  children: any
+  sidebar: SidebarState
+  position?: SidebarStateOptions['position']
+  hidden?: boolean
+}
+
+export function SidebarProvider({
+  children,
+  position,
+  hidden,
+  sidebar,
+}: SidebarProviderProps) {
+  useSubscribable(sidebar)
 
   React.useEffect(() => {
     if (typeof hidden !== 'undefined') {
-      cms.sidebar.hidden = hidden
+      sidebar.hidden = hidden
     }
   }, [hidden])
 
+  // TODO: Uncomment this
+  // if (sidebar.hidden) return children
```

Comment:
@spbyrne I think uncommenting this might solve #954?

Reply:
I'll have to test, but it seems that way! That would be a big improvement.

Reply:
Works as expected 👍